test/e2e_node/node_shutdown_linux_test.go

```diff
@@ -62,24 +62,38 @@ var _ = SIGDescribe("GracefulNodeShutdown [Serial] [NodeFeature:GracefulNodeShut
 	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
 
 	ginkgo.BeforeEach(func() {
-		if _, err := exec.LookPath("systemd-run"); err == nil {
-			if version, verr := exec.Command("systemd-run", "--version").Output(); verr == nil {
-				// sample output from $ systemd-run --version
-				// systemd 245 (245.4-4ubuntu3.13)
-				re := regexp.MustCompile(`systemd (\d+)`)
-				if match := re.FindSubmatch(version); len(match) > 1 {
-					systemdVersion, err := strconv.Atoi(string(match[1]))
-					if err != nil {
-						framework.Logf("failed to parse systemd version with error %v, 'systemd-run --version' output was [%s]", err, version)
-					} else {
-						// See comments in issue 107043, this is a known problem for a long time that this feature does not work on older systemd
-						// https://github.com/kubernetes/kubernetes/issues/107043#issuecomment-997546598
-						if systemdVersion < 245 {
-							e2eskipper.Skipf("skipping GracefulNodeShutdown tests as we are running on an old version of systemd : %d", systemdVersion)
-						}
-					}
-				}
-			}
+		if _, err := exec.LookPath("systemd-run"); err != nil {
+			e2eskipper.Skipf("skipping GracefulNodeShutdown tests as we are running on a non-systemd system: %v", err)
+		}
+		version, err := exec.Command("systemd-run", "--version").Output()
+		if err != nil {
+			e2eskipper.Skipf("skipping GracefulNodeShutdown tests as we are running on a non-systemd system: %v", err)
+		}
+
+		// sample output from $ systemd-run --version
+		// systemd 245 (245.4-4ubuntu3.13)
+		re := regexp.MustCompile(`systemd (\d+)`)
+		match := re.FindSubmatch(version)
+		if len(match) <= 1 {
+			e2eskipper.Skipf("skipping GracefulNodeShutdown tests as we are running on a non-systemd system: version %s", version)
+		}
+
+		systemdVersion, err := strconv.Atoi(string(match[1]))
+		if err != nil {
+			e2eskipper.Skipf("skipping failed to parse systemd version with error %v, 'systemd-run --version' output was [%s]", err, version)
+		}
+
+		// See comments in issue 107043, this is a known problem for a long time that this feature does not work on older systemd
+		// https://github.com/kubernetes/kubernetes/issues/107043#issuecomment-997546598
+		if systemdVersion < 245 {
+			e2eskipper.Skipf("skipping GracefulNodeShutdown tests as we are running on an old version of systemd : %d", systemdVersion)
+		}
+
+		framework.Logf("Running systemd version %d", systemdVersion)
+
+		err = checkInhibit()
```

Comment:
After I keep trying long time, I found that this can avoid flaky, I'm not quite sure why it can work, I guess it can wait for the dbus to be ready?